### PR TITLE
Activity log framework

### PR DESCRIPTION
**Pull request by massimocianfroccaPA**
[IG-2000](https://hdruk.atlassian.net/browse/IG-2000)

### CODE REVIEW

**Comment by RobinKavPA** [comment]
@massimocianfroccaPA - I think the PR should be against the activity log branch rather than 2.2dev as we haven't merged it into dev for the work that will be released before activity log

File: src/resources/activitylog/activitylog.model.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { model, Schema } from 'mongoose';
+
+const ActivityLogSchema = new Schema({
+	eventType: { type: String, required: true },
```
Comment (RobinKavPA) [inline src/resources/activitylog/activitylog.model.js:4]: We can set the allowed event types here based on the constants object you added.  This way the mongoose validation will allow only the expected types found in the enumerator.  

eventType: { type: String, required: true, **enum: Object.values(constants.activityLogEvents)** },

File: src/resources/activitylog/activitylog.model.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { model, Schema } from 'mongoose';
```
Comment (RobinKavPA) [inline src/resources/activitylog/activitylog.model.js:1]: Based on comment below you'll need to import your constant object here also.

File: src/resources/activitylog/activitylog.model.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	eventType: { type: String, required: true },
+	userTypes: [],
+	timestamp: { type: Date, required: true },
+	user: { type: Schema.Types.ObjectId, ref: 'User', required: true },
+	versionId: { type: Schema.Types.ObjectId, required: true },
+	version: { type: String, required: true },
+	plainText: { type: String, required: true },
+	detailedText: String,
+	html: { type: String, required: true },
+	detailedHtml: String,
```
Comment (RobinKavPA) [inline src/resources/activitylog/activitylog.model.js:13]: This is just a general comment to talk through during PR meeting, I think this model works for all requirements as well as giving us the strings for downloading the activity log without HTML.  We can tweak this as we go if needed however.

**Review by RobinKavPA — COMMENTED** [review]
Looks like great work to me, nicely extendible and in keeping with our existing practices.  This gives me plenty to go on for the GET API.

**Comment by massimocianfroccaPA** [comment]
With new changes introduced by Robin this is obsolete. Other PRs will merge this content.

**Pull request closed without merging**